microsoft/testsuites/core/storage.py

```diff
@@ -0,0 +1,203 @@
+        os_disk_partition = self._get_os_disk_partition(log, node)
+        if os_disk_partition == "sda":
+            mntresource = "/dev/sdb1 " + resource_disk_mount_point
+        else:
+            mntresource = "/dev/sda1 " + resource_disk_mount_point
+
+        output = node.tools[Cat].run("/etc/mtab").stdout
+        if mntresource in output:
+            log.debug("ResourceDisk entry is present.")
+        else:
+            raise LisaException("ResourceDisk entry is not present.")
+
+    @TestCaseMetadata(
+        description="""
+        This test will check that the swap is correctly configured on the VM.
+        Steps:
+        1. Check if swap file/partition is configured by checking the output of
+        `swapon -s` and `lsblk`.
+        2. Check swap status in `waagent.conf`.
+        3. Verify that values in step 1 and step 2 are equal.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            environment_status=EnvironmentStatus.Deployed,
+        ),
+    )
+    def verify_swap(self, log: Logger, node: RemoteNode) -> None:
+        swapon_output = node.tools[SwapOn].run("-s", sudo=True).stdout
+        lsblk_output = node.tools[Lsblk].run().stdout
+        _is_swap_enabled = self._is_swap_enabled(log, node)
+
+        if (
+            ("swap" in swapon_output) or ("SWAP" in lsblk_output)
+        ) and not _is_swap_enabled:
+            raise LisaException(
+                "Swap is disabled in waagent.conf but swap partition/files found."
+            )
+
+        if (
+            ("swap" not in swapon_output) and ("SWAP" not in lsblk_output)
+        ) and _is_swap_enabled:
+            raise LisaException(
+                "Swap is enabled in waagent.conf but no swap partition/files found."
+            )
+
+    def _get_wala_conf_path(self, node: RemoteNode) -> str:
+        if isinstance(node.os, CoreOs):
+            return "/usr/share/oem/waagent.conf"
+        elif node.os.information.vendor == "clear-linux-os":
+            return "/usr/share/defaults/waagent/waagent.conf"
+        else:
+            return "/etc/waagent.conf"
+
+    def _get_data_disk_mount_point(
+        self,
+        log: Logger,
+        node: RemoteNode,
+    ) -> str:
+        if node.shell.exists(
+            PurePosixPath("/var/log/cloud-init.log")
+        ) and node.shell.exists(PurePosixPath("/var/lib/cloud/instance")):
+            log.debug("Data handled by cloud-init.")
+            mount_point = "/mnt"
+        else:
+            log.debug("ResourceDisk handled by waagent.")
+            walacfg_path = self._get_wala_conf_path(node)
+            walacfg = node.tools[Cat].run(walacfg_path).stdout
+            matched = self._resource_disk_mount_point_regex.fullmatch(walacfg)
+            assert matched
+            mount_point = matched.group("resource_disk_mount_point")
+        return mount_point
+
+    def _get_os_disk_partition(
```

Comment:
This method can be in disk feature, so it's shareable.

Reply:
Do you mean in `\lisa\features\disks.py` ? Also, how about placing this in `mount` tool folder? It's processing this value from the output of mount command

Reply:
The partition information may be used in other test cases. The mount is more like the `mount` command. If we put it there, it's not easy to find the method.